### ISSUE TITLE
No module named 'imp'

### ISSUE THREAD

**Issue by Les1ie** [comet-ml/issue-tracking#545]
## Describe the Bug
No module named 'imp'

## Expected behavior
Just run it.

## Where is the issue?
- [x] Comet Python SDK
- [ ] Comet UI
- [ ] Third Party Integrations (Huggingface, TensorboardX, Pytorch Lighting etc)

## To Reproduce
Platform: `Python 3.12.2` on `Windows`.
comet_ml: `comet_ml/noarch::comet_ml-3.1.4-py_0`

## Stack Trace
If possible please include the full stack trace of your issue here

```shell
python /Workspace/demo/Pytorch_light_example.py
```

```
Traceback (most recent call last):
  File "e:\Workspace\demo\Pytorch_light_example.py", line 17, in <module>
    import comet_ml
  File "C:\Users\DELL\.conda\envs\pyg_cpu\Lib\site-packages\comet_ml\__init__.py", line 62, in <module>
    from .monkey_patching import CometModuleFinder
  File "C:\Users\DELL\.conda\envs\pyg_cpu\Lib\site-packages\comet_ml\monkey_patching.py", line 19, in <module>
    import imp
ModuleNotFoundError: No module named 'imp'
```

**Comment by jverre** [comment]
Hi,
It seems like you are using an old version of the Python SDK (`3.1.4`) that was released in 2020. Can you try updating the SDK to the lastest version (`3.42.0`) ?

**Comment by Les1ie** [comment]
Hey, I'm using python 3.12, the latest version of python~🤣
It seems that the 'imp' is removed in this version.



---Original---
From: "Jacques ***@***.***&gt;
Date: Tue, May 14, 2024 17:58 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [comet-ml/issue-tracking] No module named 'imp' (Issue #545)




 
Hi,
 It seems like you are using an old version of the Python SDK (3.1.4) that was released in 2020. Can you try updating the SDK to the lastest version (3.42.0) ?

**Comment by dsblank** [comment]
@Les1ie , indeed, we don't have a conda version ready for Python 3.12 yet. In the meantime, you should be able to get by with "pip install comet_ml". Or, alternatively, use Python 3.10.

**Comment by dsblank** [comment]
Reported internally as CM-10385.

**Comment by Lothiraldan** [comment]
@dsblank Hello, we just released the version 3.42.1 of the SDK with a conda package for Python 3.12

**Comment by Les1ie** [comment]
That's great.
